packages/google_maps_flutter/google_maps_flutter/example/integration_test/google_maps_test.dart

```diff
@@ -22,6 +22,28 @@ void main() {
   IntegrationTestWidgetsFlutterBinding.ensureInitialized();
   GoogleMapsFlutterPlatform.instance.enableDebugInspection();
 
+  // Repeatedly checks an asynchronous value against a test condition, waiting
+  // one frame between each check, returing the value if it passes the predicate
+  // before [maxTries] is reached.
+  //
+  // Returns null if the predicate is never satisfied.
+  //
+  // This is useful for cases where the Maps SDK has some internally
+  // asynchronous operation that we don't have visibility into (e.g., native UI
+  // animations).
+  Future<T?> waitForValueMatchingPredicate<T>(WidgetTester tester,
+      Future<T> Function() getValue, bool Function(T) predicate,
+      {int maxTries = 100}) async {
```

Comment:
Does it make sense to keep track of the number of tries to see if this number is ideal? Is 100 enough that it should 'fix' this issue?

Reply:
Locally it was <5 when I tested a few times, I just picked something pretty large to avoid flake (but not so large that it'll be really slow when it would actually fail). If we see flake we can up it.